lineman/app/models/group_model.coffee

```diff
@@ -148,6 +148,12 @@ angular.module('loomioApp').factory 'GroupModel', (DraftableModel, AppConfig) ->
         @remove()
         _.each @memberships(), (m) -> m.remove()
 
+    markAsRead: =>
+      @remote.postMember(@key, 'mark_as_read')
+      _.map @discussions(), (discussion) ->
+        return unless discussion.isUnread()
+        discussion.update(lastReadAt: moment(), readSalientItemsCount: discussion.salientItemsCount)
```

Comment:
Actually the more I look at this the more I don't like it. We're getting out of sync with the server here, don't have a good way of handling failure, and are left with weirdo guessed-at values designed to make `isUnread()` false, which don't get updated because we're not responding with anything useful from the server. We also don't update other attributes like `readCommentsCount` and `readItemsCount` here, meaning they're out of sync on the record itself.

I think we need to just wait for the request to come back, perhaps marking the threads with something like `readOnClient` which could exclude them from the unread count in the meantime.